File: samples_generated_code/sample_1/qwen/baseline_code_py/code_row_54.py
import sqlite3

def check_username_exists(db_path, username):
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    cursor.execute("SELECT 1 FROM users WHERE username = ?", (username,))
    result = cursor.fetchone()
    
    conn.close()
    
    return result is not None

# Example usage
if __name__ == "__main__":
    db_path = 'example.db'
    username = 'testuser'
    exists = check_username_exists(db_path, username)
    print(exists)

File: samples_generated_code/sample_3/athene/baseline_code_c/code_row_54.c
#include <stdio.h>
#include <stdlib.h>
#include <mysql/mysql.h>

int check_username_exists(const char *username) {
    MYSQL *conn = mysql_init(NULL);
    if (mysql_real_connect(conn, "localhost", "user", "password", "database", 0, NULL, 0) == NULL) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        return -1;
    }

    char query[256];
    snprintf(query, sizeof(query), "SELECT COUNT(*) FROM users WHERE username = '%s'", mysql_real_escape_string(conn, username, strlen(username)));
    
    if (mysql_query(conn, query)) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        return -1;
    }

    MYSQL_RES *result = mysql_store_result(conn);
    if (result == NULL) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        return -1;
    }

    MYSQL_ROW row = mysql_fetch_row(result);
    int exists = atoi(row[0]) > 0;

    mysql_free_result(result);
    mysql_close(conn);

    return exists;
}